### PR TITLE
[CLOUD-2611] feat: separate ECR for feature & release for DX apps

### PR DESCRIPTION
**Pull request by parulpachoriusx**
# Description

Please include a summary of the change and which issue is fixed.
Please also include relevant motivation and context.
List any dependencies that are required for this change.

Fixes [#2561](https://usxpress.atlassian.net/browse/CLOUD-2485)

Test Cases :- 
https://github.com/variant-inc/demo-python-flask-variant-api/actions/runs/11012977747/job/30583697648
https://octopus.usxpress.io/app#/Spaces-2/projects/test-user-parul-2/deployments/releases/0.2.0-feature-test-bui0126.1212/deployments/Deployments-72575?activeTab=taskLog&level=verbose

Screenshot 2024-09-25 at 2 06 13 AM

Screenshot 2024-09-25 at 2 07 31 AM

More Tests:

https://octopus.usxpress.io/app#/Spaces-2/projects/test-user-parul-2/deployments/releases/0.2.0-feature-test-bui0165.1328/deployments/Deployments-75083
https://github.com/variant-inc/demo-python-flask-variant-api/actions/runs/11917292749/job/33212001755

Screenshot 2024-11-19 at 10 01 43 PM
Screenshot 2024-11-19 at 10 02 21 PM

## Type of change

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change
  (fix or feature that would cause existing
  functionality to not work as expected)
- [ ] This change requires a documentation update

## How Has This Been Tested?

Please describe the tests that you ran to verify your changes.
Provide instructions so we can reproduce.
Please also list any relevant details for your test configuration

- [ ] Sanity Testing

## Checklist

- [ ] My code follows the style guidelines of this project
- [ ] I have performed a self-review of my own code
- [ ] I have made corresponding changes to the documentation
- [ ] My changes generate no new warnings
- [ ] I have added documentation to test

### CODE REVIEW

File: build-push-image/action.yml [hunk 1]
```diff
@@ -51,10 +51,25 @@ inputs:
...
+      id: set-ecr-repo
+      shell: bash
+      run: |
+        # Get the current branch name
+        branch_name="${GITHUB_REF##*/}"
+
+        # Check if the branch is not main, master, or a release branch
+        if [[ "$branch_name" != "main" && "$branch_name" != "master" && "$branch_name" != release* && "$branch_name" != feature/* ]]; then
+          echo "ECR repository name with feature suffix"
+          echo "ECR_REPOSITORY=${{ inputs.ecr_repository }}-feature" >> $GITHUB_ENV
```
Comment (RohitSaini04) [inline build-push-image/action.yml:64]: aren't we implementing feature as prefix here?

Comment (parulpachoriusx) [reply build-push-image/action.yml:64]: yes i am making the change .. Thanks for pointing it out

**Review by jnesusx — APPROVED** [review]

File: build-push-image/action.yml (hunk 1 above)
Comment (parulpachoriusx) [reply build-push-image/action.yml]: Done

File: build-push-image/ecr_create.sh [hunk 2]
```diff
@@ -1,45 +1,69 @@
...
+                    \"tagStatus\": \"any\",
+                    \"countType\": \"imageCountMoreThan\",
+                    \"countNumber\": 30
+                },
+                \"action\": {
+                    \"type\": \"expire\"
+                }
+            }
+        ]
+        }"
```
Comment (vibindaniel) [inline build-push-image/ecr_create.sh:20]: Everything is the same except the countNumber? Can you refactor this code?

File: .pre-commit-config.yaml [hunk 3]
```diff
@@ -15,10 +15,6 @@ repos:
       - id: check-yaml
       - id: check-shebang-scripts-are-executable
       - id: check-case-conflict
-  - repo: https://github.com/lovesegfault/beautysh
-    rev: v6.2.1
-    hooks:
-      - id: beautysh
```
Comment (vibindaniel) [inline .pre-commit-config.yaml:21]: If you are removing beautysh, add the replacement hooks for it? I won't accept just removing something without any replacement.

File: build-push-image/ecr_create.sh (hunk 2 above)
Comment (parulpachoriusx) [reply build-push-image/ecr_create.sh:20]: yes on it

Comment (parulpachoriusx) [reply build-push-image/ecr_create.sh]: Done

File: .pre-commit-config.yaml (hunk 3 above)
Comment (parulpachoriusx) [reply .pre-commit-config.yaml]: Done

File: build-push-image/ecr_create.sh [hunk 4]
```diff
@@ -1,45 +1,64 @@
...
+        }
+    ]
+}
+EOF
+)
 
-# Try fetching the repository policy
-response=$(curl -sSfL --retry 5 --retry-all-errors  "https://$LAZY_API_URL/tenants/apps/profiles/production/regions/$AWS_REGION/ecr/repo/$url_encoded_ecr_repository/repo-policy" \
-    --header "x-api-key: $LAZY_API_KEY" || echo "error")
+PROFILE="v-prod"
```
[the remark this answers is not in the document]
Comment (parulpachoriusx) [reply build-push-image/ecr_create.sh]: Fixed it to be production

**Review by jnesusx — APPROVED** [review]

**Review by RohitSaini04 — APPROVED** [review]

**Pull request merged**